sdk/core/Azure.Core/src/Request.cs

```diff
@@ -2,48 +2,128 @@
+        public new virtual RequestMethod Method
+        {
+            get => RequestMethod.Parse(MethodCore);
+            set => MethodCore = value.Method;
+        }
 
         /// <summary>
         /// Gets or sets the request content.
         /// </summary>
-        public virtual RequestContent? Content { get; set; }
+        public new virtual RequestContent? Content
+        {
+            get => (RequestContent?)ContentCore;
+            set => ContentCore = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the client request id that was sent to the server as <c>x-ms-client-request-id</c> headers.
+        /// </summary>
+        public abstract string ClientRequestId { get; set; }
+
+        /// <summary>
+        /// Gets the request HTTP headers.
+        /// </summary>
+        public new RequestHeaders Headers => new(this);
+
+        #region Overrides for "Core" methods from the PipelineRequest Template pattern
+
+        /// <summary>
+        /// Gets or sets the value of <see cref="PipelineRequest.Method"/> on
+        /// the base <see cref="PipelineRequest"/> type.
+        /// </summary>
+        protected override string MethodCore
+        {
+            get => _method;
+            set => _method = value;
+        }
+
+        /// <summary>
+        /// Gets or sets the value of <see cref="PipelineRequest.Uri"/> on
+        /// the base <see cref="PipelineRequest"/> type.
+        /// </summary>
+        protected override Uri? UriCore
+        {
+            // The _uriBuilder field on this type is the source of truth for
+            // the type's Uri implementation. Accessing it through the Uri
+            // property allows us to reuse the lazy-instantation implemented
+            // there.
+            get => Uri.ToUri();
+
+            // This setter effectively adapts the BCL Uri type to the Azure.Core
+            // RequestUriBuilder interface, in that the only way
+            // RequestUriBuilder provides to fully reset the Uri (i.e. from null)
+            // is to create a new instance of the builder.
+            set
+            {
+                if (value is null)
+                {
+                    Uri = new();
+                }
+                else
+                {
+                    Uri.Reset(value);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the value of <see cref="PipelineRequest.Content"/> on
+        /// the base <see cref="PipelineRequest"/> type.
+        /// </summary>
+        protected override BinaryContent? ContentCore
+        {
+            get => _content;
+            set => _content = (RequestContent?)value;
+        }
+
+        /// <summary>
+        /// Gets the value of <see cref="PipelineRequest.Headers"/> on
+        /// the base <see cref="PipelineRequest"/> type.
+        /// </summary>
+        protected override PipelineRequestHeaders HeadersCore
```

Comment:
Can we leave this abstract? In general, can whatever is abstract in Azure.Core today should stay abstract.

Reply:
Discussed offline